lib/storage/file.js

```diff
@@ -796,6 +796,162 @@ File.prototype.getSignedUrl = function(options, callback) {
+ *     when this policy will expire.
+ * @param {object[][]=} options.equals - Array of request parameters and 
+ *     their expected value (e.g. [["$<field>", "<value>"]]). Values are 
+ *     translated into equality constraints in the conditions 
+ *     field of the policy document (e.g. ["eq", "$<field>", "<value>"]).
+ *     If only one equality condition is to be specified, options.equals 
+ *     can be a one-dimensional array (e.g. ["$<field>", "<value>"])
+ * @param {object[][]=} options.startsWith - Array of request parameters and 
+ *     their expected prefixes (e.g. [["$<field>", "<value>"]]). Values are 
+ *     translated into starts-with constraints in the conditions field
+ *     of the policy document (e.g. ["starts-with", "$<field>", "<value>"])
+ *     If only one prefix condition is to be specified, options.startsWith 
+ *     can be a one-dimensional array (e.g. ["$<field>", "<value>"])
+ * @param {string=} options.acl - ACL for the object from possibly predefined 
+ *     ACLs
+ * @param {string=} options.successRedirect - The URL to which the user 
+ *     client is redirected if the upload is successfull
+ * @param {string=} options.successStatus - The status of the Google Storage 
+ *     response if the upload is successfull (must be string)
+ * @param {object=} options.contentLengthRange - Object providing 
+ *     minimum (options.contentLengthRange.min) and maximum 
+ *     (options.contentLengthRange.max) value for the request's 
+ *     content length
+ *
+ * @example
+ * file.getSignedPolicy({
+ *   equals: ["$Content-Type", "image/jpeg"],
+ *   contentLengthRange: {min: 0, max: 1024},
+ *   expiration: Math.round(Date.now() / 1000) + (60 * 60 * 24 * 14) // 2 weeks.
+ * }, function(err, policy) {
+ *     // policy.string: the policy document, plain text
+ *     // policy.base64: the policy document, base64
+ *     // policy.signature: the policy signature, base64
+ * });
+ */
+File.prototype.getSignedPolicy = function(options, callback) {
+  if (options.expiration < Math.floor(Date.now() / 1000)) {
+    throw new Error('An expiration date cannot be in the past.');
+  }
+
+  var expirationString = new Date(options.expiration).toISOString();
+  var conditions = [
+    ['eq', '$key', this.name],
+    {
+      bucket: this.bucket.name
+    },
+  ]; 
+
+  if (util.is(options.equals, 'array')) {
+    if (options.equals.length === 2 && 
+      util.is(options.equals[0], 'string') && 
+      util.is(options.equals[1], 'string')) {
+      conditions.push(
+        ['eq', options.equals[0], options.equals[1]]
+      );
+    } else {
+      options.equals.forEach(function(condition) {
+        if (!util.is(condition, 'array') || condition.length !== 2) {
+          throw new Error(
+            'Equals condition must be an array of 2 elements.'
+          );
+        } 
+        conditions.push(['eq', condition[0], condition[1]]);
+      });
+    }
+  }
```

Comment:
To handle supporting the `[]` and `[][]` thing, this should make it a little less noisy:

```js
if (util.is(options.condition, 'array')) {
  if (!util.is(options.condition[0], 'array')) {
    options.condition = [options.condition];
  }

  options.condition.forEach(function(condition) {
    // ...
  });
}
```